### DISCUSSION TITLE
Text truncation with TextFlow solved in GMF context, bug what about GEF solely?

### DISCUSSION THREAD

**Discussion by Groundation** [eclipse-gef/gef-classic#597] [Q&A] [answered]
Found that the bug with text being cut off at the end of line is solved way back in 2010: 

https://bugs.eclipse.org/bugs/show_bug.cgi?id=235549

but I am still to decipher the solution (newbie to Eclipse and GEF & Draw2D especially) but it seems it's been solved for the solutions based on GMF. In my application I only use GEF and Draw2D.

Can anybody give me a helping hand with understanding the solved issue and how I can apply the solution for my context (I would've expected the solution would be integrated with the GEF since then)? 

Nenad

**Comment by azoitl** [comment, marked as answer]
What we found out is that using the `ScaledGraphics` class for drawing your figures. We added some time back the option to disable the use of `ScaledGraphics` when creating your rooteditpart. See the constructor of `ScalableFreeformRootEditPart` for example. I hope this helps.

**Reply by Groundation** [reply]
Greetings @azoitl ! Thanks very much for your prompt answer. 

I have just disabled `ScaledGraphics` in my editor when setting the root edit part `GraphApp.setRootEditPart(new ScalableFreeformRootEditPart(false))` as you suggested and indeed the text is not being truncated anymore! Thanks! Was getting quite frustrated with this one. 

I do however see some very subtle changes in the way the things are drawn without ScaledGraphics when zoom in, but I doubt it will affect the editors interface in any major way. 

Much thanks.

**Reply by azoitl** [reply]
Great that this is the solution. Yes the drawing is bit different but how I experienced it in Eclipse 4diac the normal graphics looks better and cleaner.
